File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    corrects = [0, 0, 0]
    supo1 = [1,2,3,4,5]
    supo2 = [2,1,2,3,2,4,2,5]
    supo3 = [3,3,1,1,2,2,4,4,5,5]
    
    for i, answer in enumerate(answers):
        if answer == supo1[i % len(supo1)]:
            corrects[0] += 1
        if answer == supo2[i % len(supo2)]:
            corrects[1] += 1
        if answer == supo3[i % len(supo3)]:
            corrects[2] += 1
    
    maxScore = max(corrects)
    return [i + 1 for i, c in enumerate(corrects) if c == maxScore]

File: 프로그래머스/lv1/42840. 모의고사/모의고사.js
function solution(answers) {
    const one = [1,2,3,4,5],
          two = [2,1,2,3,2,4,2,5],
          three = [3,3,1,1,2,2,4,4,5,5];
    const corrects = [0, 0, 0];
    
    for (let i = 0; i < answers.length; i++) {
        corrects[0] += one[i % 5] === answers[i] ? 1 : 0;
        corrects[1] += two[i % 8] === answers[i] ? 1 : 0;
        corrects[2] += three[i % 10] === answers[i] ? 1 : 0;
    }
    
    const answer = []
    const first = Math.max(...corrects);
    for (let i = 0; i < 3; i++) {
        if (corrects[i] === first) {
            answer.push(i + 1);
        }
    }
    return answer;
}